### PR TITLE
VideoPress: reload preview on oembed fail

### PR DESCRIPTION
**Pull request by thedebian**
Fixes Automattic/greenhouse#936

#### Changes proposed in this Pull Request:

Regenerate oembed preview on VideoPress oembed failure.
This fixes the problem with the video embed width/height not being known before generating the oembed preview. 

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* On a sandboxed wpcom and Jetpack with VideoPress available
* Apply wpcom patch D70588-code
* Create a new post on your Jetpack
* Add a /video block and upload a portrait video
* ✅ The video block should generate the embed with the right aspect ratio (you can check out the greenhouse issue to see screenshots of previous failure)

Note: Sometimes, it works on the first try, so you may have to try multiple times until you see a core/video being displayed instead of a VideoPress video and you should see the block being reloaded at a rate between 2 to 10s. depending on the try count.

### CODE REVIEW

**Comment by roundhill** [comment]
> so you may have to try multiple times until you see a core/video being displayed instead of a VideoPress video

I hit this a lot, it's a little odd. Is there some way we could show a placeholder or something if it hits this case?

**Comment by thedebian** [comment]
> > so you may have to try multiple times until you see a core/video being displayed instead of a VideoPress video
> 
> I hit this a lot, it's a little odd. Is there some way we could show a placeholder or something if it hits this case?

I'm afraid we only have two choices right now : let the oembed return the video itself (like now) and display the video OR make the oembed return an HTTP error and display the "oops there's no content" screen (don't remember exactly what the screen says, but it's the screen with the broken file icon in it when a video is missing), as the whole request is handled by the core MediaPlaceholder block, not by us directly

**Comment by roundhill** [comment]
> I'm afraid we only have two choices right now : let the oembed return the video itself (like now) and display the video OR make the oembed return an HTTP error and display the "oops there's no content" screen (don't remember exactly what the screen says, but it's the screen with the broken file icon in it when a video is missing), as the whole request is handled by the core MediaPlaceholder/embed block, not by us directly

Hmm, ok. I'm not sure which experience is better 🤷. Maybe it's just my jurassic tube site but it takes quite a while for the correct embed to show, and getting the double video block settings is really jarring.

File: projects/plugins/jetpack/extensions/blocks/videopress/edit.js [hunk 1]
```diff
@@ -56,6 +56,8 @@ const VideoPressEdit = CoreVideoEdit =>
 				isUpdatingRating: false,
 			};
 			this.posterImageButton = createRef();
+			this.timerID = null;
```
Comment (roundhill) [inline projects/plugins/jetpack/extensions/blocks/videopress/edit.js:59]: `timerId`? Or maybe something like `previewCacheTimer`

**Review by roundhill — APPROVED** [review]
I merged master in, hope that's ok :D I wanted to test it on wp.com instead of my slow Jurassic Tube site. The experience seemed much better, less of the awkward in-between state.

Just had a variable naming thought.

File: projects/plugins/jetpack/extensions/blocks/videopress/edit.js (hunk 1 above)
Comment (thedebian) [reply projects/plugins/jetpack/extensions/blocks/videopress/edit.js]: 👍 Renamed "previewCacheReloadTimer"

**Comment by thedebian** [comment]
Hi Jetpack team ! There's a ✅ from @roundhill (requested changes have been done and I just forgot to talk about this PR after our last team call), but he's AFK for a while; I changed the status to "Needs Review" so we could have your review and move forward, I'll triple-check with him when he's back before merging :) (There's also a linked wpcom PR which is ready to go ✅  D70588 )

**Review by jeherve — APPROVED** [review]

**Comment by thedebian** [comment]
Changeset 237518

**Pull request merged**